utility/types/message.go

```diff
@@ -0,0 +1,434 @@
+package types
+
+import (
+	"bytes"
+	"log"
+	"net/url"
+	"strconv"
+	"strings"
+
+	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/types"
+	"google.golang.org/protobuf/proto"
+)
+
+type Message interface {
+	proto.Message
+
+	SetSigner(signer []byte)
+	ValidateBasic() types.Error
+}
+
+var _ Message = &MessageStakeApp{}
+
+func (msg *MessageStakeApp) ValidateBasic() types.Error {
+	if err := ValidateAmount(msg.Amount); err != nil {
+		return err
+	}
+	if err := ValidatePublicKey(msg.PublicKey); err != nil {
+		return err
+	}
+	if err := ValidateRelayChains(msg.Chains); err != nil {
+		return err
+	}
+	return ValidateOutputAddress(msg.OutputAddress)
+}
+
+func (msg *MessageStakeApp) SetSigner(signer []byte) {
+	msg.Signer = signer
+}
+
```

Comment:
Some decorative comments to indicate that we are extending a different message type would be visually more comforting, something like:

```
// MessageEditStakeApp
```